### ISSUE TITLE
stopEvent not working in shadow-root

### ISSUE THREAD

**Issue by n0ps** [openlayers/openlayers#10162]
**Describe the bug**
With an OpenLayers map inside shadow-root (WebComponent) Overlay with `stopEvent = true` does not stop events from propagating.

Problem is possibly in: https://github.com/openlayers/openlayers/blob/a3c137cff15ae8027d0cdb04a162ebb2fa977451/src/ol/PluggableMap.js#L937-L943

For click events originating from an Overlay div events have their first `mapBrowserEvent.target` to be `shadow-root` and therefore `if (target.parentElement === this.overlayContainerStopEvent_) {` never evaluates to `true`.

Is this expected behavior in v6.x or does it exists a workaround that works inside shadow-root?

**To Reproduce**
1. Create a minimal HTML-page with a WebComponent and minimal OL map.
```
<html>

<head>
    <script src="https://cdn.jsdelivr.net/gh/openlayers/openlayers.github.io@master/en/v6.0.1/build/ol.js"></script>
</head>

<body>
    <ol-map></ol-map>

    <script>
        class OLComponent extends HTMLElement {
            connectedCallback() {
                this.shadow = this.attachShadow({ mode: 'open' });
                this.render();
            }

            displayMap(div, overlayDiv) {
                const layer = new ol.layer.Tile({
                    source: new ol.source.OSM()
                });

                const map = new ol.Map({
                    layers: [layer],
                    target: div,
                    view: new ol.View({
                        center: [0, 0],
                        zoom: 2
                    })
                });

                const overlay = new ol.Overlay({
                    stopEvent: true,
                    insertFirst: true,
                    element: overlayDiv
                });

                map.addOverlay(overlay);

                map.on('singleclick', function (e) {
                    overlay.setPosition(e.coordinate);
                    overlay.element.innerHTML = '<div style="padding: 10px; background: #fff;">Testing</div>';
                });
            }

            render() {
                const div = document.createElement('div');
                div.classList.add('map');
                div.style.width = '600px';
                div.style.height = '300px';
                this.shadow.appendChild(div);

                const link = document.createElement('link');
                link.rel = 'stylesheet';
                link.type = 'text/css';
                link.href = 'https://cdn.rawgit.com/openlayers/openlayers.github.io/master/en/v5.3.0/css/ol.css';
                this.shadow.appendChild(link);

                const overlay = document.createElement('div');
                this.shadow.appendChild(overlay);

                this.displayMap(div, overlay);
            }
        }

        customElements.define('ol-map', OLComponent);

    </script>
</body>

</html>
```
2. Singleclicks on the map will display Overlay at correct positions. **But if clicking the overlay div, events are propagated to the map despite being told not to.** This happens not in v5.3.0.

**Expected behavior**
Should work like in v5.3.0.

**Comment by fredj** [comment]
'singleclick' and 'click' events are emulated by OpenLayers; that's probably the reason of this issue.
Using a pointer event (for example 'pointerdown') works as expected

**Comment by n0ps** [comment]
Thanks for answer. With Pointer events it works like expected, but implementing logic on client side detecting different kind of 'clicks' is quite a complicated workaround for this issue.

Will the actual problem with emulation be fixed to work as in v5.3.0?

**Comment by ahocevar** [comment]
> Will the actual problem with emulation be fixed to work as in v5.3.0?

We definitely accept a pull request to fix this. To me, it is not clear what causes the issue, so I'm unable to provide a fix myself.

**Comment by fredj** [comment]
To emulate the events, the `pointerdown` event is [stored](https://github.com/openlayers/openlayers/blob/v6.0.1/src/ol/MapBrowserEventHandler.js#L194) and then used in the [`emulateClick_` function](https://github.com/openlayers/openlayers/blob/v6.0.1/src/ol/MapBrowserEventHandler.js#L162)

The problem is that the target is changed in between (Event retargeting), see:
https://developers.google.com/web/fundamentals/web-components/shadowdom#events

A possible solution could be to use the [`composedPath()` function](https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath) instead of the `parentElement`'s of the `target`.
